apps/login/src/app/api/loginname/route.ts

```diff
@@ -37,6 +46,91 @@ export async function POST(request: NextRequest) {
             return NextResponse.json(error, { status: 500 });
           });
       } else {
+        const loginSettings = await getLoginSettings(organization);
+        // TODO: check if allowDomainDiscovery has to be allowed too, to redirect to the register page
+        // user not found, check if register is enabled on organization
+
+        if (
+          loginSettings?.allowRegister &&
+          !loginSettings?.allowUsernamePassword
+        ) {
+          // TODO redirect to loginname page with idp hint
+          const identityProviders = await getActiveIdentityProviders(
+            organization,
+          ).then((resp) => {
+            return resp.identityProviders;
+          });
+
+          if (identityProviders.length === 1) {
+            const host = request.nextUrl.origin;
+
+            const provider =
+              identityProviders[0].type === IdentityProviderType.GITHUB
+                ? "github"
+                : identityProviders[0].type === IdentityProviderType.GOOGLE
+                  ? "google"
+                  : identityProviders[0].type === IdentityProviderType.AZURE_AD
+                    ? "azure"
+                    : identityProviders[0].type === IdentityProviderType.SAML
+                      ? "saml"
+                      : identityProviders[0].type === IdentityProviderType.OIDC
+                        ? "oidc"
+                        : "oidc";
```

Comment:
Why not a switch and throw an error if none matches?

Reply:
true.. changed 👍